proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,349 @@
+\end{codeblock}
+
+\setcounter{Paras}{0}
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        for (; first != last; (void)++result, ++first)
+          ::new (static_cast<void*>(addressof(*result)))
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@(std::move(*first));
+        return @\changed{last}{\{last, result + (last - first)\}}@;
+\end{codeblock}
+
+\pnum
+\remarks If an exception is thrown, some objects in \tcode{[first, last)} are left in a valid, but unspecified state.
+
+{\color{addclr}
+\pnum
+\requires \tcode{result} shall not be in the range \tcode{[first, last)}.
+
+\begin{codeblock}
+  template <InputRange Rng, OutputIterator O>
+    requires Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>> &&
+             Same<value_type_t<O>&, reference_t<O>> &&
+             @\newtxt{MoveConstructible<value_type_t<O>{>}()}@
+  tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>>
+    uninitialized_move(Rng&& rng, O&& result);
+\end{codeblock}
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        return uninitialized_move(begin(rng), end(rng), std::forward<O>(result));
+\end{codeblock}
+
+\pnum
+\remarks If an exception is thrown, some objects in \tcode{[first, last)} are left in a valid, but unspecified state.
+
+\pnum
+\requires \tcode{result} shall not be in the range \tcode{[first, last)}.
+} %% \color{addclr}
+
+\begin{codeblock}
+  template <@\changed{class}{InputIterator}@ @\changed{InputIterator}{I}@, @\changed{class Size, class ForwardIterator}{OutputIterator O}@>
+    @\added{requires Constructible<value_type_t<O>, reference_t<I>{>}() \&\&}@
+             @\added{Same<value_type_t<O>\&, reference_t<O>{>}}@
+             @\newtxt{MoveConstructible<value_type_t<O>{>}()}@
+  @\changed{pair<InputIterator, OutputIterator>}{tagged_pair<tag::in(I), tag::out(O)>}@
+    uninitialized_move_n(@\changed{InputIterator}{I}@&& first, @\changed{Size}{difference_type_t<I>}@ n, @\changed{ForwardIterator}{O}@&& out)
+\end{codeblock}
+
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        for (; n > 0; ++result, (void) ++first, --n)
+          ::new (static_cast<void*>(addressof(*result)))
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@(std::move(*first));
+        return {first,result};
+\end{codeblock}
+
+\pnum
+\remarks If an exception is thrown, some objects in \tcode{[first, next(first, n))} are left in a valid, but unspecified state.
+
+{\color{addclr}
+\pnum
+\requires \tcode{result} shall not be in the range \tcode{[first, next(first, n))}.
+} %%\color{addclr}
+
+\rSec3[uninitialized.fill]{uninitialized_fill}
+\begin{codeblock}
+  template <@\changed{class}{CopyConstructible}@ @\changed{ForwardIterator}{T}@, @\changed{class}{OutputIterator<const T\&>}@ @\changed{T}{O},@
+            @\added{Sentinel<O> S}@>
```

Comment:
Yes, it's ok that `fill` and `uninitialized_fill` have differing template parameter order.